lib/Controller/PasswordController.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/**
+ * @author Jörn Friedrich Dreyer <jfd@butonic.de>
+ *
+ * @copyright Copyright (c) 2018, ownCloud GmbH
+ * @license GPL-2.0
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\PasswordPolicy\Controller;
+
+use OCP\AppFramework\Controller;
+use OCP\AppFramework\Http\RedirectResponse;
+use OCP\AppFramework\Http\TemplateResponse;
+use OCP\Authentication\IAccountModuleController;
+use OCP\IConfig;
+use OCP\IRequest;
+use OCP\ISession;
+use OCP\IURLGenerator;
+use OCP\IUser;
+use OCP\IUserManager;
+use OCP\IUserSession;
+
+class PasswordController extends Controller implements IAccountModuleController {
+
+	/** @var IUser */
+	protected $user;
+	/** @var IUserManager */
+	protected $userManager;
+	/** @var IConfig */
+	protected $config;
+	/** @var ISession */
+	protected $session;
+	/** @var IURLGenerator */
+	protected $urlGenerator;
+
+	/**
+	 * PasswordController constructor.
+	 *
+	 * @param $appName
+	 * @param IRequest $request
+	 * @param IUserSession $userSession
+	 * @param IUserManager $userManager
+	 * @param IConfig $config
+	 * @param ISession $session
+	 * @param IURLGenerator $urlGenerator
+	 */
+	public function __construct($appName, IRequest $request, IUserSession $userSession, IUserManager $userManager, IConfig $config, ISession $session, IURLGenerator $urlGenerator) {
+		parent::__construct($appName, $request);
+		$this->user = $userSession->getUser(); // might be null
```

Comment:
no - get the user in the controller method - the user is set to the session ins some cases in the sec middleware